### ISSUE TITLE
Bug: calcite-modal cannot have many Calcite Components as its firstFocus property

### ISSUE THREAD

**Issue by patrickarlt** [Esri/calcite-components#1375]
### Summary

`<calcite-modal>` has a `firstFocus` attribute that allows you to configure which element should receive focus when the modal open. This is great except that many Calcite components do not have a `focus` method. So you end up not being able to set focus to say a `<calcite-checkbox>` or `<calcite-button>` inside a `<calcite-modal>`.

### Actual Behavior

A quick audit of what I need for my app yields several different methods of setting focus to a Calcite component. None of these work because they are not `focus()` which is expected by `<calcite-modal>` https://github.com/Esri/calcite-components/blob/master/src/components/calcite-modal/calcite-modal.tsx#L212-L223

* `<calcite-checkbox>` - uses a `focused` attribute.
* `<calcite-button>` -  has a `setFocus()` method.

### Expected Behavior

Setting `firstFocus` on `<calcite-modal>` should work for other Calcite Components.

### Reproduction Steps

https://codepen.io/patrickarlt/pen/VwKjzxd

**Comment by driskull** [comment]
@macandcheese this should be somewhat easy. We just have to wrap the `isFocusable` method to check if the element has a `setFocus()` method and return true as well.

**Comment by noahmulfinger** [comment]
Not sure if this should be broken off into a separate issue, but I'm also running into focus issues when using `calcite-tile-select` in a `calcite-modal`. There's no way to programmatically focus `calcite-tile-select`, so the `calcite-modal` ends up focusing on an element after it in the DOM.

**Comment by driskull** [comment]
@noahmulfinger both `calcite-tile-select` and `calcite-tile` would need `setFocus` methods added to them. Can you create an issue for that?

**Comment by driskull** [comment]
Closing. If you see a problem feel free to reopen @patrickarlt .
